platform/ios/src/MGLMapView.mm

```diff
@@ -1061,29 +1056,56 @@ - (void)installAttributionButtonConstraints {
     [self updateConstraintsForOrnament:self.attributionButton
                            constraints:self.attributionButtonConstraints
                               position:self.attributionButtonPosition
-                                  size:[self sizeForOrnament:self.attributionButton constraints:self.attributionButtonConstraints]
+                       sizeConstraints:[self sizeConstraintsForOrnament:self.attributionButton constraints:self.attributionButtonConstraints]
                                margins:self.attributionButtonMargins];
 }
 
-- (CGSize)sizeForOrnament:(UIView *)view
-              constraints:(NSMutableArray *)constraints {
-    // avoid regenerating size constraints
-    CGSize size;
-    if(constraints && constraints.count > 0) {
-        for (NSLayoutConstraint * constraint in constraints) {
-            if([constraint.identifier isEqualToString:@"width"]) {
-                size.width = constraint.constant;
-            }
-            else if ([constraint.identifier isEqualToString:@"height"]) {
-                size.height = constraint.constant;
-            }
-        }
+- (NSArray *)sizeConstraintsForOrnament:(UIView *)view
+                            constraints:(NSMutableArray *)constraints {
+
+    NSLayoutConstraint *widthConstraint = nil;
+    NSLayoutConstraint *heightConstraint = nil;
+
+    if (view == _scaleBar) {
+        widthConstraint = [NSLayoutConstraint constraintWithItem:_scaleBar
+                                                       attribute:NSLayoutAttributeWidth
+                                                       relatedBy:NSLayoutRelationEqual
+                                                          toItem:self
+                                                       attribute:NSLayoutAttributeWidth
+                                                      multiplier:.5
```

Comment:
Is half the width of the view essentially the maximum size possible? How does this impact running on iPad?

Reply:
It may be a bit long on the iPad, but I think it's acceptable 😂